beacon-chain/powchain/log_processing.go

```diff
@@ -76,25 +76,23 @@ func (w *Web3Service) ProcessDepositLog(depositLog gethTypes.Log) {
 
 	// We then decode the deposit input in order to create a deposit object
 	// we can store in our persistent DB.
+	validData := true
 	depositInput, err := helpers.DecodeDepositInput(depositData)
 	if err != nil {
 		log.Errorf("Could not decode deposit input %v", err)
-		return
+		validData = false
 	}
 
 	deposit := &pb.Deposit{
 		DepositData:     depositData,
 		MerkleTreeIndex: index,
 	}
 
-	validData := true
-
 	// Make sure duplicates are rejected pre-chainstart.
-	if !w.chainStarted {
+	if !w.chainStarted && validData {
```

Comment:
I dont know if we should be rejecting invalid deposits, pre-chainstart. I guess we can leave it for now for the testnet, but its something to keep mindful of in the future.